packages/drivers/odsp-driver/src/prefetchLatestSnapshot.ts

```diff
@@ -0,0 +1,228 @@
+import {
+    createOdspLogger,
+    fetchAndParseAsJSONHelper,
+    getOdspResolvedUrl,
+    getWithRetryForTokenRefresh,
+    IOdspResponse,
+} from "./odspUtils";
+import { IOdspSnapshot } from "./contracts";
+
+/**
+ * Function to prefetch the snapshot and cached it in the persistant cache, so that when the container is loaded
+ * the cached latest snapshot could be used and removes the network call from the critical path.
+ * @param resolvedUrl - Resolved url to fetch the snapshot.
+ * @param getStorageToken - function that can provide the storage token for a given site. This is
+ *  is also referred to as the "VROOM" token in SPO.
+ * @param persistedCache - Cache to store the fetched snapshot.
+ * @param logger - Logger to have telemetry events.
+ * @param hostSnapshotFetchOptions - Options to fetch the snapshot if any. Otherwise default will be used.
+ * @returns - True if the snapshot is cached, false otherwise.
+ */
+export async function prefetchLatestSnapshot(
+    resolvedUrl: IResolvedUrl,
+    getStorageToken: TokenFetcher<OdspResourceTokenFetchOptions>,
+    persistedCache: IPersistedCache = new LocalPersistentCache(),
+    logger: ITelemetryBaseLogger,
+    hostSnapshotFetchOptions: ISnapshotOptions | undefined,
+): Promise<boolean> {
+    const odspLogger = createOdspLogger(ChildLogger.create(logger, "PrefetchSnapshot"));
+    const odspResolvedUrl = getOdspResolvedUrl(resolvedUrl);
+    const snapshotUrl = odspResolvedUrl.endpoints.snapshotStorageUrl;
+    let snapshotOptions: ISnapshotOptions = {
+        deltas: 1,
+        channels: 1,
+        blobs: 2,
+    };
+
+    if (hostSnapshotFetchOptions) {
+        snapshotOptions = {...snapshotOptions, ...hostSnapshotFetchOptions};
+    }
+
+    const storageTokenFetcher = toInstrumentedOdspTokenFetcher(
+        odspLogger,
+        odspResolvedUrl,
+        getStorageToken,
+        true /* throwOnNullToken */,
+    );
+
+    const snapshotCacheEntry: ICacheEntry = {
+        type: snapshotKey,
+        key: "",
+        file: {
+            resolvedUrl: odspResolvedUrl,
+            docId: odspResolvedUrl.hashedDocumentId,
+        },
+    };
+
+    return getWithRetryForTokenRefresh(async (tokenFetchOptions) => {
+        // No need to ask cache twice - if first request was unsuccessful, cache unlikely to have data on second turn.
+        if (tokenFetchOptions.refresh) {
+            await fetchSnapshot(
+                snapshotUrl,
+                odspResolvedUrl,
+                snapshotOptions,
+                tokenFetchOptions,
+                storageTokenFetcher,
+                odspLogger,
+                persistedCache,
+                snapshotCacheEntry,
+            );
+        } else {
+            await PerformanceEvent.timedExecAsync(
+                odspLogger,
+                { eventName: "ObtainSnapshot" },
+                async (event: PerformanceEvent) => {
+                    const cachedSnapshot: ISnapshotCacheValue | undefined =
```

Comment:
I think consulting cache should not be here. It goes against the reason this function exists - to prefetch latest data from storage. Even if we keep it, there is on need to push same bits back into cache on lines 118+
That said, I can see a case where there might be two different usages:
1. Only fetch from storage
2. check cache and fetch storage only if there is nothing in cache.
If we want to support both, we need to consult with HostStoragePolicy.concurrentSnapshotFetch

Reply:
I was thinking of second scenario, where we check cache and if not present then only fetch from storage. This is because the cache could already have the snapshot, so fetching it again will be redundant. 
Also we are not fetching concurrently if I am getting it right. It is like if not present in cache then only fetch from storage,

Reply:
Yes but if we are thinking that it is to find latest snapshot data, then I can avoid checking the cache and get it from storage.

Reply:
I think most of the users would want to refresh their cache even if it exists. Meeting scenario in particular calls out using this functionality 5 min before meeting to warm up the state. So it's not that interesting if cache has some value from day before.
That said, other scenarios might be find with using earlier cached value and prefer spend less resources now. It depends, but I think only consulting cache is not enough. Only doing fetch is better, but might turn out limiting as well, so it's better  to leave policy choice to a caller

Reply:
Changed it to always fetch snapshot.